packages/caliper-ethereum/lib/ethereum.js

```diff
@@ -161,25 +171,32 @@ class Ethereum extends BlockchainInterface {
      */
     async sendTransaction(context, contractID, contractVer, methodCall, timeout) {
         let status = new TxStatus();
+        let params = {from: context.fromAddress}
         try {
             context.engine.submitCallback(1);
             let receipt = null;
             let methodType = 'send';
             if (methodCall.isView) {
                 methodType = 'call';
+            } else {
+                let nonce = context.nonces[context.fromAddress]
+                context.nonces[context.fromAddress] = nonce + 1
+                params.nonce = nonce
             }
             if (methodCall.args) {
-                receipt = await context.contracts[contractID].methods[methodCall.verb](...methodCall.args)[methodType]({from: context.fromAddress});
+                params.gas = 1000 + await context.contracts[contractID].methods[methodCall.verb](...methodCall.args).estimateGas()
```

Comment:
It's a remove call.  This too can be hoisted to a global context, however the margin of error will need to be raised because of the weird zero-memory incentives the gas charged is variable in movement tests.

Reply:
Well, if it's needed, then it should stay here, reflects a valid scenario. Unless the callback specifies an explicit value, or a default value is set in the config (again, todo), you have to make this call.